interfaces/matlab/CMakeLists.txt

```diff
@@ -7,229 +7,371 @@
     else()
+        list(APPEND CMAKE_INSTALL_RPATH "@loader_path")
+        list(APPEND CMAKE_INSTALL_RPATH "@loader_path/../${CMAKE_INSTALL_LIBDIR}")
+        list(APPEND CMAKE_INSTALL_RPATH "@executable_path")
+        list(APPEND CMAKE_INSTALL_RPATH "@executable_path/../${CMAKE_INSTALL_LIBDIR}")
+    endif()
+endif()
 
-        file(GLOB SHARED_LIB_HEADERS
-             ${CMAKE_SOURCE_DIR}/src/helics/shared_api_library/*.h)
-
-        # custom command for building the wrap file
-        add_custom_command(
-            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/helicsMEX.cpp
-            COMMAND
-                "${SWIG_EXECUTABLE}"
-                "-matlab"
-                "-c++"
-                -o
-                "helicsMEX.cpp"
-                "-I${CMAKE_SOURCE_DIR}/src/helics/shared_api_library"
-                ${CMAKE_CURRENT_SOURCE_DIR}/helicsMatlab.i
-            DEPENDS
-                ../helics.i ${CMAKE_CURRENT_SOURCE_DIR}/helicsMatlab.i
-                ${SHARED_LIB_HEADERS} ${CMAKE_CURRENT_SOURCE_DIR}/matlab_maps.i
-        )
+###############################
+# Find required dependencies
+###############################
+# Find HELICS
+if(NOT hasParent OR NOT TARGET HELICS::helicsSharedLib)
+    find_package(HELICS 2 REQUIRED)
+endif()
 
-        # add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/helicsMEX.cpp COMMAND
-        # ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_SOURCE_DIR}/modifyOctSourceFile.cmake
-        # DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/helicsMEX.cpp )
+if(HELICS_ENABLE_SWIG OR HELICS_SWIG_GENERATE_INTERFACE_FILES_ONLY)
+    # Find SWIG
+    if(NOT DEFINED SWIG_FOUND OR SWIG_VERSION VERSION_LESS 3)
+        find_package(SWIG 3 REQUIRED)
+        include(${SWIG_USE_FILE})
+    endif()
 
-        install(
-            DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/+helics
-            DESTINATION matlab
-            COMPONENT matlab
-        )
-        install(
-            FILES
-                ${CMAKE_CURRENT_BINARY_DIR}/SwigGet.m
-                ${CMAKE_CURRENT_BINARY_DIR}/SwigMem.m
-                ${CMAKE_CURRENT_BINARY_DIR}/SwigRef.m
-            DESTINATION matlab
-            COMPONENT matlab
-        )
-        install(
-            FILES
-                ${CMAKE_CURRENT_SOURCE_DIR}/pisender.m
-                ${CMAKE_CURRENT_SOURCE_DIR}/pireceiver.m
-                ${CMAKE_CURRENT_SOURCE_DIR}/GetFullPath.m
-            DESTINATION matlab
-            COMPONENT matlab
+    # check to make sure the swig executable supports matlab
+    execute_process(
+        COMMAND ${SWIG_EXECUTABLE} -help
+        OUTPUT_VARIABLE SWIG_HELP_OUTPUT
+    )
+    string(FIND "${SWIG_HELP_OUTPUT}" "-matlab" MATLAB_HELP_FOUND)
+    if(${MATLAB_HELP_FOUND} LESS 0)
+        set(MATLAB_SWIG_NOT_AVAILABLE 1)
+        message(
+            WARNING
+                " SWIG VERSION does not support Matlab, reverting to build only"
         )
     endif()
 
+    # Find helics.i swig interface file
```

Comment:
is this the same code we use in the other interface modules?  if so it something that could be extracted to a separate file or would that make things more complicated?

Reply:
It would be convenient to have it in a separate file, but is more complicated. It needs to be accessible for building as a subproject of HELICS and when building standalone (ideally without the assumption that it is part of the HELICS source tree).